doc/perception_configuration/src/cylinder_segment.cpp

```diff
@@ -0,0 +1,121 @@
+#include <ros/ros.h>
+#include <sensor_msgs/PointCloud2.h>
+#include <pcl_conversions/pcl_conversions.h>
+#include <pcl/filters/passthrough.h>
+#include <pcl/filters/extract_indices.h>
+#include <pcl/features/normal_3d.h>
+#include <pcl/segmentation/sac_segmentation.h>
+
+ros::Publisher pub;
+
+void cloud_cb(const sensor_msgs::PointCloud2ConstPtr& input)
```

Comment:
This function is going to need quite a bit of cleanup to be included in the tutorials. One of the goals of the tutorial demo scripts is to demonstrate best ROS/MoveIt!/C++ practices. This function is very long and does too many things. This needs to be refactored with more comments.